lightning/src/ln/channelmanager.rs

```diff
@@ -7937,6 +7941,8 @@ where
 			return Err(Bolt12SemanticError::UnsupportedChain);
 		}
 
+		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(self);
```

Comment:
Might be a bit better to put this after all the potential error paths/right before we push into `self.pending_offers_messages`, but that's a very minor optimization.